### ISSUE TITLE
Inference Code

### ISSUE THREAD

**Issue by eschaffn** [SaadNoor555/Document-Layout-Analysis#1]
Hey there, would it be possible to share the code for inferencing with the trained model?

I can't figure out how to get the outputs (bounding boxes and labels).

Thank you!

**Comment by SaadNoor555** [comment]
Here is an inference script using the models: https://www.kaggle.com/code/qoobeeshy/rusty-inference
It has a lot of post processing done and doesn't explicitly output the bounding box but if you take your time understanding the code you will get an idea. If you need help feel free to ask, I will provide a simple script to output masks, label and bounding boxes

**Comment by eschaffn** [comment]
So I was able to output the 6 value tensor of the model, which I assume is [Left, Upper, Height, Width, Confidence, Label]? This should be enough for me to figure out the rest, however I was wondering what the classes map to. I tried to figure it out from the data preprocessing code but wasn't able to fully piece it together.

Is it:

{0: text,
 1: title,
 2: list,
 3: table,
 4: figure}

I notice the configs only include labels from 0-3, did you remove a label here?

**Comment by SaadNoor555** [comment]
The model used here was not fine-tuned on publynet dataset which is why the labels are not 0-4, it was fine-tuned on [badlad](https://www.kaggle.com/competitions/dlsprint2/data?select=badlad) dataset. That is why it only has 0-3 labels. labels are: 
`{
    0: paragraph,
    1: textbox,
    2: image,
    3: table
}
`
None of the weights we have distributed [here](https://www.kaggle.com/datasets/qoobeeshy/yoloft) have 5 classes. 
However, we have a 50 epoch weight that we have not uploaded that was trained on a single fold of publynet dataset, which matches your described classes. If you need it contact us.

**Comment by eschaffn** [comment]
> The model used here was not fine-tuned on publynet dataset which is why the labels are not 0-4, it was fine-tuned on [badlad](https://www.kaggle.com/competitions/dlsprint2/data?select=badlad) dataset. That is why it only has 0-3 labels. labels are: `{ 0: paragraph, 1: textbox, 2: image, 3: table } ` None of the weights we have distributed [here](https://www.kaggle.com/datasets/qoobeeshy/yoloft) have 5 classes. However, we have a 50 epoch weight that we have not uploaded that was trained on a single fold of publynet dataset, which matches your described classes. If you need it contact us.

I see. Thank you for the clarification! The publynet trained model would be great if you don't mind sharing!

**Comment by eschaffn** [comment]
> The model used here was not fine-tuned on publynet dataset which is why the labels are not 0-4, it was fine-tuned on [badlad](https://www.kaggle.com/competitions/dlsprint2/data?select=badlad) dataset. That is why it only has 0-3 labels. labels are: `{ 0: paragraph, 1: textbox, 2: image, 3: table } ` None of the weights we have distributed [here](https://www.kaggle.com/datasets/qoobeeshy/yoloft) have 5 classes. However, we have a 50 epoch weight that we have not uploaded that was trained on a single fold of publynet dataset, which matches your described classes. If you need it contact us.

![image](https://github.com/SaadNoor555/Document-Layout-Analysis/assets/94634191/00af1361-f237-4b8d-8002-38ec599f325e)
I see this model here https://www.kaggle.com/datasets/qoobeeshy/yoloft?select=epoch75_publy.pt would it be an img model or aug model?

**Comment by SaadNoor555** [comment]
The particular model you mentioned was pretraiend on a fold of publynet dataset then finetuned on badlad dataset. It also outputs four classes. 
All the models that have "img" in their name are only trained to segment image class in a document, that is because a model that works on multiple classes tends to leave holes in image and table class, so a specialized model was trained to segment image class only. 
All the classes that have "aug" in their name are trained on some sort of augmented image. They output for 4 classes as well.
You can read our paper to understand the whole process [here](https://arxiv.org/abs/2309.00848)
And we are working on retrieving the publynet finetuned weight, we'll let you know as soon as we publish it.

**Comment by eschaffn** [comment]
Thank you!

**Comment by SaadNoor555** [comment]
Uploaded the publynet trained weight [here](https://www.kaggle.com/datasets/qoobeeshy/yoloft?select=publy.pt)
check it out!
